examples/check_integration_error.py

```diff
@@ -6,123 +6,131 @@
 × t2 × vd × repeats) w/ normally distributed random noise.
 Check that the following match:
 
-- integral w/ error (the canned routine :func:`~pyspecProcScripts.integral_w_errors`)
+- integral w/ error (the canned routine
+  :func:`~pyspecProcScripts.integral_w_errors`)
 - propagate error based off the programmed σ of the normal distribution
 - set the error bars based on the standard deviation (along the repeats
   dimension) of the *real* part of the integral
-- propagate error based off the variance of the noise in the inactive
-  coherence channels (do this manually inside this script -- should mimic
-  what :func:`~pyspecProcScripts.integral_w_errors` does)
+- propagate error based off the variance of the noise in the inactive coherence
+  channels (do this manually inside this script -- should mimic what
+  :func:`~pyspecProcScripts.integral_w_errors` does)
 """
-from pylab import *
-from pyspecdata import *
-from pyspecProcScripts import integrate_limits, integral_w_errors
+import pyspecdata as psd
+import numpy as np
+from numpy import r_
+from pyspecProcScripts import integral_w_errors, select_pathway
+from pylab import rcParams
+import sympy as s
+from collections import OrderedDict
 
+rcParams["image.aspect"] = "auto"  # for sphinx gallery
 # sphinx_gallery_thumbnail_number = 1
 
-init_logging(level="debug")
-fl = figlist_var()
-t2 = nddata(r_[0:1:1024j], "t2")
-vd = nddata(r_[0:1:40j], "vd")
-ph1 = nddata(r_[0, 2] / 4.0, "ph1")
-ph2 = nddata(r_[0:4] / 4.0, "ph2")
+t2, vd, ph1, ph2 = s.symbols("t2 vd ph1 ph2")
+n_repeats = 10
+noise_bounds = (0, 200)  # seem reasonable to me
 signal_pathway = {"ph1": 0, "ph2": 1}
-excluded_pathways = [(0, 0), (0, 3)]
-# this generates fake clean_data w/ a T₂ of 0.2s
-# amplitude of 21, just to pick a random amplitude
-# offset of 300 Hz, FWHM 10 Hz
-clean_data = (
-    21 * (1 - 2 * exp(-vd / 0.2)) * exp(+1j * 2 * pi * 100 * t2 - t2 * 10 * pi)
+# {{{ Generate fake IR dataset with multiple repeats
+# This generates fake clean_data w/ a T₂ of 0.2s amplitude of 21, just to pick
+# a random amplitude offset of 300 Hz, FWHM 10 Hz
+expression = (
+    21
+    * (1 - 2 * s.exp(-vd / 0.2))
+    * s.exp(+1j * 2 * s.pi * 100 * t2 - t2 * 10 * s.pi)
 )
-clean_data *= exp(signal_pathway["ph1"] * 1j * 2 * pi * ph1)
-clean_data *= exp(signal_pathway["ph2"] * 1j * 2 * pi * ph2)
+orderedDict = [
+    ("vd", psd.nddata(r_[0:1:40j], "vd")),
+    ("ph1", psd.nddata(r_[0, 2] / 4.0, "ph1")),
+    ("ph2", psd.nddata(r_[0:4] / 4.0, "ph2")),
+    ("t2", psd.nddata(r_[0:1:1024j], "t2")),
+]
+clean_data = psd.fake_data(
+    expression, OrderedDict(orderedDict), signal_pathway, scale=0
+)
+clean_data.reorder(["vd", "t2"], first=False)
```

Comment:
no longer using manual generation of fake data since we have a function that does that for us